### ISSUE TITLE
[binding-websockets] Current implementation missing certain functionality

### ISSUE THREAD

**Issue by miguelrk** [eclipse/thingweb.node-wot#476]
**TLDR:** currently, `binding-websockets` does not handle properties, actions, nor meta-interactions

**In general, I believe `node-wot` should strive to provide an equal, complete implementation across all protocols (despite their intrinsic differences). In other words, [this table](https://github.com/eclipse/thingweb.node-wot/issues/213) should have as many checks as possible.**

I'm aware that the necessity of a WebSocket subprotocol in order to wrap websocket messages for the WebSocketServer to be able to route/handle incoming requests has been previously [discussed](https://github.com/eclipse/thingweb.node-wot/issues/128#issuecomment-545796340). Having said that, I'm not sure if the current `binding-websockets` implementation is in standby because of that or the missing features are simply pending to be implemented.

What appears to be missing (in comparisson with http and mqtt bindings at least, note some might actually not be desired):

1) **`ws://{host}:{port}/`** should return list of exposed things (`binding-mqtt` I think does not publish this list on initial connection. The `binding-http` though does indeed allow making a GET request to `http://{host}:{port}/` to fetch it)
2) **`ws://{host}:{port}/{title}`** should return the TD of the exposed thing (similar to how `binding-mqtt` publishes td on intial subscription to `{title}/#` topic, `binding-http` handles this via normal GET request to `http://{host}:{port}/{title}`)
3) **`ws://{host}:{port}/{title}/properties/{name}`** should handle readproperty, writeproperty, observeproperty, unobserveproperty
4) **`ws://{host}:{port}/{title}/actions/{name}`** should handle invokeaction
5) **`ws://{host}:{port}/{title}/all/properties`** should handle readallproperties, writeallproperties, observeallproperties, unobserveallproperties
6) **`ws://{host}:{port}/{title}/multiple/properties`** should handle readmultipleproperties and writemultipleproperties (NOTE: this might be done already, I couldn't find out if `/multiple/properties` or where?)
7) **other meta interactions** e.g. subscribeallevents, unsubscribeallevents `/all/events` but I think those have not yet been implemented in node-wot

Regarding the websocket subprotocol (might be more advanced discussions here already), something like this:
```json
{
    "type": "writemultipleproperties",
    "data": {
        "temperature": 19,
        "speed": 18
    }
}
```
might be everything the server/client needs to correctly distinguish/identify messages. 

But maybe a more complete, even standards-based implementation like [cloudevents](https://cloudevents.io/) might be worth it:
```json
{
  "specversion": "1.0",
  "type": "writemultipleproperties",
  "id": "xxxx-xxxx-xxxx",
  "source": "bigco.com",
   "data": {
        "temperature": 19,
        "speed": 18
    }
}
```

I see that the `binding-mqtt` is somewhat more complete (e.g. it already handles properties and actions). I could maybe provide a PR if similar functionality (and that mentioned in 1-7 above) is indeed desired for `binding-websockets` as well.

Thanks!

**Comment by miguelrk** [comment]
Update: I actually did just open-up a [PR](https://github.com/eclipse/thingweb.node-wot/pull/480) into the v0.8 branch to get this going in case you want to have a look.

**Comment by egekorkan** [comment]
We can still support multiple subprotocols. This is invisible for consumer applications anyway but for Things built via node-wot, it would imply that the Thing code needs a way to choose a subprotocol. This issue will exist if sse would be implemented on the server side too.

**Comment by danielpeintner** [comment]
> Update: I actually did just open-up a [PR](https://github.com/eclipse/thingweb.node-wot/pull/480) into the v0.8 branch to get this going in case you want to have a look. It's a WIP still, will continue to work on this and keep you updated...

@miguelrk's PR https://github.com/eclipse/thingweb.node-wot/pull/521 helped to improve the situation (by adding handlers for properties and actions).

 Do we think more should be added like all or multiple operations?
@relu91 @egekorkan
